### PR TITLE
MBQL sum-where

### PR DESCRIPTION
**Pull request by sbelak**
Adds `sum-where` MBQL clause with the following semantics:

`[:sum-where concrete-field filter-clause]`

- [x] Update the docs

### CODE REVIEW

File: modules/drivers/druid/src/metabase/driver/druid/query_processor.clj [hunk 1]
```diff
@@ -477,7 +478,12 @@
   ([pred output-name] (ag:filtered (parse-filter pred)
                                    (ag:count output-name))))
 
-(defn- create-aggregation-clause [output-name ag-type ag-field]
+(defn- ag:sumWhere
+  ([field pred output-name] (ag:filtered (parse-filter pred)
+                                         (ag:doubleSum field output-name))))
```
Comment (camsaul) [inline modules/drivers/druid/src/metabase/driver/druid/query_processor.clj]: since there's only one arity to this fn might as well remove the extra `()`

**Review by camsaul — APPROVED** [review]
works for me

File: modules/drivers/druid/src/metabase/driver/druid/query_processor.clj (hunk 1 above)
Comment (sbelak) [reply modules/drivers/druid/src/metabase/driver/druid/query_processor.clj]: Agree. I already have another PR that cleans up the druid code and makes it compliant with our style (#9757 -- but it's 90% just white space things, so probably not worth the review)

**Comment by kurumpa** [comment]
It's not implemented on the query builder side yet, is it?

**Comment by sbelak** [comment]
@kurumpa: no, it's not

**Comment by kurumpa** [comment]
@sbelak Any chance to see this in the builder soon? Thanks a lot for contributing, btw

**Comment by sbelak** [comment]
We're hoping to included with the new query builder in 0.33

**Pull request merged**